staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -33,6 +33,14 @@ const (
 	// StreamingProxyRedirects controls whether the apiserver should intercept (and follow)
 	// redirects from the backend (Kubelet) for streaming requests (exec/attach/port-forward).
 	StreamingProxyRedirects utilfeature.Feature = "StreamingProxyRedirects"
+
+	// owner: timstclair
+	// alpha: v1.7
+	//
+	// AdvancedAuditing enables a much more general API auditing pipeline, which includes support for
+	// pluggable output backends and an audit policy specifying how different requests should be
+	// audited.
+	AdvancedAuditing utilfeature.Feature = "AdvancedAuditing"
```

Comment:
> We could re-add the old WithAudit filter as WithBasicAudit and use that if the feature is disabled. Wdyt?

That's what I was suggesting in https://github.com/kubernetes/kubernetes/pull/45766#discussion_r116766143. Since the pipeline is growing in complexity, I think we should do that.

Reasons for adding a feature flag:

- This code is coming in hot and it mitigates risk. Additionally, it would make a code freeze exception easier if we end up needing it...
- Default audit level (off = metadata, on = none)
- If users specify flags for the new features (e.g. set a policy, different backend) they should be aware that they are using an *alpha* feature. Requiring the feature gate to be set makes that more explicit
- Makes the support & completeness level of advanced auditing explicit